src/db-api.js

```diff
@@ -68,47 +68,141 @@ const transactionsHistoryForAddresses = (db: Pool) => async (
 
 // The remaining queries should be used only for the purposes of the legacy API!
 
-/**
- * Queries DB looking for successful transactions associated with any of the given addresses.
- * @param {Db Object} db
- * @param {Array<Address>} addresses
- */
-const bulkAddressSummary = (db: Pool) => async (addresses: Array<string>): Promise<ResultSet> =>
-  db.query({
-    text: `SELECT * FROM "txs"
-      WHERE hash = ANY (SELECT tx_hash FROM "tx_addresses" WHERE address = ANY($1))
-      AND tx_state = $2
-      ORDER BY time DESC`,
-    values: [addresses, 'Successful'],
-  })
-
 /**
 * Queries TXS table looking for a successful transaction with a given hash
 * @param {Db Object} db
-* @param {*} tx
+* @param {Transaction} tx
 */
 const txSummary = (db: Pool) => async (tx: string): Promise<ResultSet> =>
   db.query({
     text: 'SELECT * FROM "txs" WHERE hash = $1 AND tx_state = $2',
     values: [tx, 'Successful'],
   })
 
+/**
+* Queries TX table looking for tx by its hash
+* @param {Db Object} db
+* @param {Transaction} tx
+*/
+const getTx = (db: Pool) => async (tx: string): Promise<ResultSet> =>
+  db.query({
+    text: 'SELECT id, block, hash::text FROM "tx" WHERE hash = $1',
+    values: [tx],
+  })
+
+/**
+* Queries BLOCK table looking for block with a given id
+* @param {Db Object} db
+* @param {Block} blockId
+*/
+const getBlockById = (db: Pool) => async (blockId: string): Promise<ResultSet> =>
+  db.query({
+    text: 'SELECT time, block_no, hash::text FROM block WHERE id = $1',
+    values: [blockId],
+  })
+
+/**
+* Quries TX* tables to get txInputs for a given transaction
+* @param {Db Object} db
+* @param {Transaction} tx
+*/
+const getTxInputs = (db: Pool) => async (tx: string): Promise<ResultSet> =>
```

Comment:
I think it would be worth typing the result especially since the db schema is not within the codebase. For example, it's not clear what the tx_out.value is supposed to be nor how the field would be named - something which a type would precisely define

Reply:
also, the parameter tx: string leaves space for doubt id it is the database identifier of the tx (in which case it should be a number I believe) or the actual hash of the tx - I would rename it to make it clear

Reply:
Something like

`declare type txInputsResult = {
    address: string,
    value: number
}`

for each db query?

Reply:
Oh I see, like this:
![image](https://user-images.githubusercontent.com/22454337/85691508-db7dd400-b6d4-11ea-9891-aae6a217ce88.png)